src/codec/decode.go

```diff
@@ -45,15 +45,15 @@ func init() {
 //
 // This implementation is inlined from https://github.com/dennwc/varint to avoid the call-site overhead
 func (cb *Buffer) DecodeVarint() (uint64, error) {
-	if cb.len == 0 {
+	if cb.Len() == 0 {
```

Comment:
I'm ... actually not 100% sure these check replacements are correct. They do pass the unit tests, and the fuzz test does fail if you remove any of them, which seems ... good?